### DISCUSSION TITLE
How to Test Out My Code in GitHub before Submitting to GitHub Classroom

### DISCUSSION THREAD

**Discussion by KS-USM** [community/community#37752] [New to GitHub]
Hello. I'm a student who had just starting out with C++ programming. My lecturer provides us the repository (eg which includes assignments/quizzes/labs) and uses autograding to automatically grade us. We can use git clone to create a clone of the repository in VS Code to do our assignments, then commit to sync the files to GitHub once we are done coding. I noticed the autograding.json file and is interested to know how can I check if my code can work before submitting to GitHub? Or is there any way for me to know my grades so in case I did something wrong in my code, I can fix my code? For now, I only know how to test my code via VS Code Terminal. Sorry that I'm very new to programming, so any help is highly appreciated! :)

**Comment by jfullstackdev** [comment]
other than that, you can ask your lecturer on that process too, maybe he/she will allow modifications or revert if your code is not working on that repo
